src/families/bitcoin/logic.ts

```diff
@@ -52,7 +56,15 @@ export const isValidRecipient = async (params: {
   if (!params.recipient) {
     return Promise.reject(new RecipientRequired(""));
   }
-  const valid = isValidAddress(params.recipient);
+
+  let valid: boolean;
+  try {
+    // Optimistically assume params.currency.id is an actual Currency
+    valid = isValidAddress(params.recipient, <Currency>params.currency.id);
+  } catch (e) {
+    // isValidAddress() will throw Error if c is not an actual Currency
+    return Promise.reject(new CurrencyNotSupported(e.message));
```

Comment:
Actually you can use `InvalidAddress` here, because it will translate appropriately on the frontend, see [here](https://github.com/LedgerHQ/ledger-live-desktop/blob/b116d532f3dc4f919ad98042b48ace8f135ec715/static/i18n/en/app.json#L3688-L3690),
where `CurrencyNotSupported`'s [translation](https://github.com/LedgerHQ/ledger-live-desktop/blob/b116d532f3dc4f919ad98042b48ace8f135ec715/static/i18n/en/app.json#L3433-L3435) would be misleading.